### PR TITLE
Remove config.Global

### PR DESCRIPTION
**Pull request by sectioneight**
This was causing lots of headaches and leaked state between tests.

GFM-105

### CODE REVIEW

**Comment by sectioneight** [comment]
@uber-go/service-framework

File: core/config/config.go [hunk 1]
```diff
@@ -175,15 +140,8 @@ func UnregisterProviders() {
...
-func InitializeGlobalConfig() {
-	_setupMux.Lock()
-	defer _setupMux.Unlock()
-	if _initialized {
-		return
-	}
-	_initialized = true
-
+// InitializeConfig initializes the ConfigurationProvider for use in a service
+func InitializeConfig() ConfigurationProvider {
```
Comment (anuptalwalkar) [inline core/config/config.go]: Lets just call it Initialize(), so that the call will be config.Initialize()

Comment (sectioneight) [inline core/config/config.go]: Great idea. Updating.

File: core/config/config.go [hunk 2]
```diff
@@ -175,15 +140,8 @@ func UnregisterProviders() {
...
-func InitializeGlobalConfig() {
-	_setupMux.Lock()
-	defer _setupMux.Unlock()
-	if _initialized {
-		return
-	}
-	_initialized = true
-
+// Initialize creates a ConfigurationProvider for use in a service
+func Initialize() ConfigurationProvider {
```
Comment (sectioneight) [inline core/config/config.go]: sure. suggestions?

**Pull request merged**